tests/unit/memefficiency.tcl

```diff
@@ -145,601 +287,323 @@ run_solo {defrag} {
                 }
-                assert {$frag < 1.1}
-                # due to high fragmentation, 100hz, and active-defrag-cycle-max set to 75,
-                # we expect max latency to be not much higher than 7.5ms but due to rare slowness threshold is set higher
-                if {!$::no_latency} {
-                    assert {$max_latency <= 30}
+                for {set j 0} {$j < $n} {incr j} { $rd read } ; # Discard replies
+            } fragment {
+                for {set j 0} {$j < $n} {incr j 2} {
+                    $rd del k$j
                 }
+                for {set j 0} {$j < $n} {incr j 2} { $rd read } ; # Discard replies
+                $rd close
             }
-            # verify the data isn't corrupted or changed
-            set newdigest [debug_digest]
-            assert {$digest eq $newdigest}
-            r save ;# saving an rdb iterates over all the data / pointers
-        } {OK}
-
-        test "Active defrag pubsub: $type" {
-            r flushdb sync
-            r config resetstat
-            r config set activedefrag no
-            r config set active-defrag-threshold-lower 5
-            r config set active-defrag-cycle-min 65
-            r config set active-defrag-cycle-max 75
-            r config set active-defrag-ignore-bytes 1500kb
-            r config set maxmemory 0
+        }
+    }
+
+    proc test_pubsub {type} {
+        set title "Active Defrag pubsub: $type"
+        test $title {
+            set n 100000
 
-            # Populate memory with interleaving pubsub-key pattern of same size
-            set n 50000
-            set dummy_channel "[string repeat x 400]"
             set rd [valkey_deferring_client]
-            set rd_pubsub [valkey_deferring_client]
+            set chan [string repeat A 50]
+
+            # https://github.com/valkey-io/valkey/issues/1774
+            # NOTE - pubsub defrag isn't working properly.  This wasn't caught before the test refactor.
+            #  This commented out code should work.  Instead, is substituted code which verifies that
+            #  defrag doesn't break the pubsub, but defrag efficacy is not verified.
+            # perform_defrag_test $title {
+            #     for {set j 0} {$j < $n} {incr j} {
+            #         set channel $chan$j
+            #         $rd subscribe $channel
+            #         $rd read ; # Discard subscribe replies
+            #         $rd ssubscribe $channel
+            #         $rd read ; # Discard ssubscribe replies
+            #     }
+            # } fragment {
+            #     for {set j 0} {$j < $n} {incr j 2} {
+            #         set channel $chan$j
+            #         $rd unsubscribe $channel
+            #         $rd read
+            #         $rd sunsubscribe $channel
+            #         $rd read
+            #     }
+            # }
+            # --- BEGIN TEMPORARY CODE ---
+            log_frag "test start (pubsub)"
```

Comment:
Will you finish this?

Reply:
No.  I created: https://github.com/valkey-io/valkey/issues/1774
I redesigned all of the defrag structure.  I rewrote all of the tests.  If I took responsibility for everything I found that doesn't work properly, I'd never stop.  I'll let some pubsub expert deep-dive to determine why it's not defragging.

Reply:
Ok, makes sense. Thanks for opening the issue.

Shall we link to this issue in the comment above then? We usually don't refer to issues in code comments, but this is a TODO that we want to track. The issue can provide more information and discussion. If someone fixes it, we want to close the issue.